lib/qs/payload_handler.rb

```diff
@@ -38,7 +37,7 @@ def run
     def run!
       processed_payload = ProcessedPayload.new
       begin
-        payload = Qs::Payload.decode(@serialized_payload)
+        payload = @daemon_data.deserialize(@serialized_payload)
```

Comment:
@jcredding again, I'll comment further in a bit, but I'd prefer to keep this `Payload` module and decode function and not have to configure, pass in and store a deserializer on the daemon/daemon data/etc.